include/HubFramework/HUBContentOperationContext.h

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright (c) 2016 Spotify AB.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import "HUBConnectivityState.h"
+
+@protocol HUBFeatureInfo;
+@protocol HUBViewModelBuilder;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ *  Protocol defining the public API of a content operation context
+ *
+ *  This API is currently only used with `HUBBlockContentOperation`. For more information
+ *  about the properties that are part of this API, refer to `HUBContentOperation`, or to
+ *  the "Content programming guide".
+ */
+@protocol HUBContentOperationContext <NSObject>
```

Comment:
Actually, I think I prefer to keep it the way it is currently implemented. The reason is that it creates a very clear separation between the API and the implementation. It clearly signals to the API user that this is not a class that you are supposed to instantiate yourself - since the initialzier is not accessible.